framework/core/buffer.cpp

```diff
@@ -89,27 +91,11 @@ VmaAllocation Buffer::get_memory() const
 
 uint8_t *Buffer::map()
```

Comment:
The `map` and `unmap` functions can be removed as they are used internally only.

The `update` function can do the copy on `mapped_data` directly if its not null.